WINGSTOP_Air_Brake_Flight_Computer_FULL_BRAKES/WINGSTOP_Air_Brake_Flight_Computer_FULL_BRAKES.ino

```diff
@@ -453,61 +497,112 @@ void read_accelerometer(){//reads and updates z_global variable
   acc = 9.81 *  -(1+mpu.getAccX()*sin(mpu.getAngleY()*3.14/180)-mpu.getAccY()*cos(mpu.getAngleY()*3.14/180)*sin(mpu.getAngleX()*3.14/180)-mpu.getAccZ()*cos(mpu.getAngleY()*3.14/180)*cos(mpu.getAngleX()*3.14/180));// uses a transformation matrix to determine global z acceleration
 }
 
-void average_acceleration_data(){//averages acceleration data over the set frequency and saves to z_global variable. Also updates t2 time variable.
+void average_acceleration_data(){//averages acceleration data over the set frequency and saves to z_global variable. Also updates t_current time variable.
   int i = 0; //iteration counting variable
   acc_avg = 0; // resets z_avg variable to 0
   
   do{ //run at least once
-    t2 = millis();        // updates t2 variable with current time
+    t_current = millis();        // updates t_current variable with current time
     read_accelerometer(); //updates z_global variable
     acc_avg += acc;    //adds all the acceleration readings for each iteration
     i++;                  //counts the number of iterations
-  }while(t2-t1 < 1000/hz-1);//exits loop once elapsed time is greater than frequency time period
+  }while(t_current-t_previous < 1000/hz-1);//exits loop once elapsed time is greater than frequency time period
 
   acc = acc_avg/i;     //averages the z acceleration data for the period
-  delta_t = t2-t1;        //update change in time variable
-  t1 = t2;                //updates the t1 point for when the loop ended
+  delta_t = t_current-t_previous;        //update change in time variable
+  t_previous = t_current;                //updates the t_previous point for when the loop ended
 }
 
 
 //***********PHYSICS FUNCTIONS***********
 
-void read_velocity(){//reads new altitude as x2 and compares to previous altitude x1 and delta_t to determine velocity
-  x2 = read_altitude(); //gets current altitude
-  vel = (x2 - x1)/(delta_t / 1000); //determines velocity
-  x1 = x2; //update previous altitude variable
+void read_velocity(){//reads new altitude as x_current and compares to previous altitude x_previous and delta_t to determine velocity
+  x_current = read_altitude(); //gets current altitude
+  vel = (x_current - x_previous)/(delta_t / 1000); //determines velocity
+  x_previous = x_current; //update previous altitude variable
+  distance_to_target = TARGET_ALTITUDE - x_current; //update distance to target. Negative value implies target has been passed.
+}
+
+void predict_apogee(){//predicts apogee with current velocity, altitude, and acceleration and sets brakes to reach target altitude
```

Comment:
I recommend renaming this function to something that makes it obvious that the air brakes are deployed here.  Maybe `predict_and_deploy()`.  I'm not the best with naming.